src/directories.cc

```diff
@@ -540,7 +540,7 @@ void Directories::add_or_update_path(const boost::filesystem::path &dir_path, co
   if(path_it==directories.end()) {
     auto g_file=Glib::wrap(g_file_new_for_path(dir_path.string().c_str())); //TODO: report missing constructor in giomm
 
-#if defined(__FreeBSD__)
+#if defined(__OpenBSD__) || defined(__NetBSD__) || defined(__FreeBSD__) || defined(__Darwin__) || defined(__DragonFly__)
```

Comment:
Thank you for the good work, I really liked the lightweight native alternative for KDevelop that I use presently. I had the same problem on DragonFlyBSD and found the commits for FreeBSD only. FreeBSD is conservative about kernel improvements so it has most ports. Other BSDs ports usually are patches upon FreeBSD ports, so the errors while building on FreeBSD usually recur on all other BSDs, unless you are working with the base system rather than ports. With the patch the IDE certainly got built on DragonFlyBSD for me.